test/unit/test_events.py

```diff
@@ -45,3 +46,29 @@ def test_formatting(self):
         # were no args passed after the initial message
         event = AdapterEventDebug(name="dbt_tests", base_msg="boop{x}boop", args=())
         self.assertTrue("boop{x}boop" in event.message())
+
+class TestEventCodes(TestCase):
+
+    def setUp(self):
+        pass
+
+    def get_all_subclasses(self, cls):
+        all_subclasses = []
+        for subclass in cls.__subclasses__():
+            all_subclasses.append(subclass)
+            all_subclasses.extend(self.get_all_subclasses(subclass))
+        return set(all_subclasses)
+
+    # checks to see if event codes are duplicated to keep codes singluar and clear.
```

Comment:
it also checks the code format with `self.assertTrue('^[A-Z][0-9]{3}', event.code)`

Reply:
added more clarifying comments.